application/src/main/java/com/sanction/thunder/dao/PilotUsersDao.java

```diff
@@ -138,11 +143,15 @@ public PilotUser update(PilotUser user) {
       table.putItem(newItem, new Expected("version").eq(oldVersion));
     } catch (ConditionalCheckFailedException e) {
       LOG.error("The user was updated while this update was in progress."
-          + " Aborting to avoid race condition.");
+          + " Aborting to avoid race condition.", e);
       throw new DatabaseException("The user to update is at an unexpected stage.",
           DatabaseError.CONFLICT);
+    } catch (AmazonServiceException e) {
```

Comment:
Can you be sure a ServiceException happens only with unsupported data?

Reply:
It's a good point, I don't think I can be. It's hard to know all the possible problems that can arise with a service exception since it seems like the main exception AWS throws when something is wrong. I can make the error message more general. Either way I want to return an Internal Service Error because it's not recoverable. Since the exception is getting logged now, we'll be able to see exactly what happened in the logs.

Reply:
OK, it should be fine for now